### PR TITLE
task3 completed

### PR DESCRIPTION
**Pull request by asintdivya123**

### CODE REVIEW

File: assignment3/src/main/java/com/asint/assignment3/model/Product.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name="PRODUCTS")
+public class Product {
+	
+	@Id
+	private int id;
+	
+	@Column(name="name")
```
Comment (StillSomehowSane) [inline assignment3/src/main/java/com/asint/assignment3/model/Product.java]: try removing the @Column or change the names to caps

File: assignment3/src/main/java/com/asint/assignment3/model/Sales.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package com.asint.assignment3.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name="SALES")
+public class Sales {
```
Comment (StillSomehowSane) [inline assignment3/src/main/java/com/asint/assignment3/model/Sales.java]: Rename to Sale

File: assignment3/src/main/java/com/asint/assignment3/model/Sales.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	
+	
+	
+	@Id
+	private int id;
+	
+	@Column(name="product_id")
+	private int product_id;
```
Comment (StillSomehowSane) [inline assignment3/src/main/java/com/asint/assignment3/model/Sales.java]: Must follow snake case so productId

File: assignment3/src/main/java/com/asint/assignment3/repository/ProductRepo.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.asint.assignment3.repository;
+
+import java.io.Serializable;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import com.asint.assignment3.model.Product;
+
+public interface ProductRepo extends JpaRepository<Product, Serializable> {
```
Comment (StillSomehowSane) [inline assignment3/src/main/java/com/asint/assignment3/repository/ProductRepo.java]: Replace Serializable

File: assignment3/src/main/java/com/asint/assignment3/service/ProductService.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		
+		//[[4,587],[6,426],[1,399],[8,280],[7,225],[2,158],[3,116],[5,66]]
+		
+		ArrayNode arrayNode = mapper.createArrayNode();
+		for(Integer[] o : data ) {
+			ObjectNode objectNode = mapper.createObjectNode();
+			
+			objectNode.put("id", o[0]);
+			
+			Optional<Product> dbProduct = productRepo.findById(o[0]);
```
Comment (StillSomehowSane) [inline assignment3/src/main/java/com/asint/assignment3/service/ProductService.java]: Remove selects from loop

File: assignment3/src/main/java/com/asint/assignment3/repository/SalesRepo.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import com.asint.assignment3.model.Sales;
+
+
+public interface SalesRepo extends JpaRepository<Sales, Serializable> {
+
+	
+	@Query(value = "select product_id, sum(quantity) as quantity from sales group by product_id order by sum(quantity) desc", nativeQuery = true)
```
Comment (StillSomehowSane) [inline assignment3/src/main/java/com/asint/assignment3/repository/SalesRepo.java]: Try removing native query

**Review by StillSomehowSane — CHANGES_REQUESTED** [review]

File: assignment3/src/main/java/com/asint/assignment3/repository/SaleRepo.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import com.asint.assignment3.model.Sale;
+
+public interface SaleRepo extends JpaRepository<Sale, Integer> {
+
+	
+	@Query("select s.productId, sum(s.quantity)*p.price, p.name from Sale s join Product p on s.productId = p.id group by s.productId order by sum(s.quantity) desc")
```
Comment (StillSomehowSane) [inline assignment3/src/main/java/com/asint/assignment3/repository/SaleRepo.java:11]: I am glad that you figured out the error without native query

**Comment by asintdivya123** [comment]
requested changes pushed @StillSomehowSane

File: assignment3/src/main/java/com/asint/assignment3/service/ProductService.java [hunk 8]
```diff
@@ -9,33 +9,35 @@
...
 
 	public ArrayNode getProductWithTotalQuanityResponse() {
-		List<Integer[]> data = salesRepo.getProductWithTotalQuanity();
+		List<Object[]> data = saleRepo.getProductWithTotalQuanity();
 		
 		//[[4,587],[6,426],[1,399],[8,280],[7,225],[2,158],[3,116],[5,66]]
 		
 		ArrayNode arrayNode = mapper.createArrayNode();
-		for(Integer[] o : data ) {
+		for(Object[] o : data ) {
```
Comment (StillSomehowSane) [inline assignment3/src/main/java/com/asint/assignment3/service/ProductService.java:29]: Good Job!

**Review by StillSomehowSane — APPROVED** [review]
Awesomeness

**Comment by asintdivya123** [comment]
Thanq

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by asintdivya123** [comment]
Yes it took time but thankfully got it

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Pull request merged**